go-controller/pkg/ovn/controller/services/loadbalancer.go

```diff
@@ -212,47 +316,54 @@ func EnsureLBs(nbClient libovsdbclient.Client, service *corev1.Service, existing
 		return group
 	}
 	for k, v := range addLBsToGroups {
-		ops, err = libovsdbops.AddLoadBalancersToGroupOps(nbClient, ops, getGroup(k), v...)
+		ops, err = libovsdbops.AddLoadBalancersToGroupOps(nbClient, ops, getGroup(k), toNBLoadBalancerList(v)...)
 		if err != nil {
 			return fmt.Errorf("failed to create ops for adding load balancers to group %s for service %s/%s: %w",
 				k, service.Namespace, service.Name, err)
 		}
 	}
 	for k, v := range removeLBsFromGroups {
-		ops, err = libovsdbops.RemoveLoadBalancersFromGroupOps(nbClient, ops, getGroup(k), v...)
+		ops, err = libovsdbops.RemoveLoadBalancersFromGroupOps(nbClient, ops, getGroup(k), toNBLoadBalancerList(v)...)
 		if err != nil {
 			return fmt.Errorf("failed to create ops for removing load balancers from group %s for service %s/%s: %w",
 				k, service.Namespace, service.Name, err)
 		}
 	}
 
 	deleteLBs := make([]*nbdb.LoadBalancer, 0, len(toDelete))
-	for uuid := range toDelete {
-		deleteLBs = append(deleteLBs, &nbdb.LoadBalancer{UUID: uuid})
+	deleteTemplates := make([]libovsdbops.TemplateMap, 0, len(toDelete))
+	for _, clb := range toDelete {
+		deleteLBs = append(deleteLBs, &nbdb.LoadBalancer{UUID: clb.UUID})
+		deleteTemplates = append(deleteTemplates, clb.Templates)
 	}
 	ops, err = libovsdbops.DeleteLoadBalancersOps(nbClient, ops, deleteLBs...)
 	if err != nil {
 		return fmt.Errorf("failed to create ops for removing %d load balancers for service %s/%s: %w",
 			len(deleteLBs), service.Namespace, service.Name, err)
 	}
 
+	ops, err = libovsdbops.DeleteChassisTemplateVarOps(nbClient, ops, deleteTemplates...)
+	if err != nil {
+		return err
+	}
+
 	recordOps, txOkCallBack, _, err := metrics.GetConfigDurationRecorder().AddOVN(nbClient, "service",
 		service.Namespace, service.Name)
 	if err != nil {
 		klog.Errorf("Failed to record config duration: %v", err)
 	}
 	ops = append(ops, recordOps...)
 
-	_, err = libovsdbops.TransactAndCheckAndSetUUIDs(nbClient, lbs, ops)
+	_, err = libovsdbops.TransactAndCheckAndSetUUIDs(nbClient, toNBLoadBalancerList(tlbs), ops)
 	if err != nil {
 		return fmt.Errorf("failed to ensure load balancers for service %s/%s: %w", service.Namespace, service.Name, err)
 	}
 	txOkCallBack()
 
 	// Store UUID of newly created load balancers for future calls.
 	// This is accomplished by the caching of LBs by the caller of this function.
-	for _, lb := range lbs {
-		wantedByName[lb.Name].UUID = lb.UUID
+	for _, tlb := range tlbs {
+		wantedByName[tlb.nbLB.Name].UUID = tlb.nbLB.UUID
 	}
 
 	klog.V(5).Infof("Deleted %d stale LBs for %#v", len(toDelete), externalIDs)
```

Comment:
Nice. I was able to verify the conversion from non-template to template lbs w/out any issues.
```
I0301 22:21:10.267795      42 loadbalancer.go:369] Deleted 3 stale LBs for map[string]string{"k8s.ovn.org/kind":"Service", "k8s.ovn.org/owner":"default/webserver"}
```